spec/index.bs

```diff
@@ -4894,18 +4894,28 @@ must be well nested.
 <dl dfn-type=method dfn-for=GPUCommandEncoder>
     : <dfn>writeTimestamp(querySet, queryIndex)</dfn>
     ::
+        Writes a timestamp value into |querySet| when all previous commands have completed executing.
 
         <div algorithm=GPUCommandEncoder.writeTimestamp>
             **Called on:** {{GPUCommandEncoder}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPUCommandEncoder/writeTimestamp(querySet, queryIndex)">
-                querySet:
-                queryIndex:
+                querySet: The query set that will store the timestamp values.
+                queryIndex: The index of the query in the query set.
             </pre>
 
             **Returns:** {{undefined}}
 
+            If any of the following conditions are unsatisfied, generate a validation error and stop.
+            <div class=validusage>
+                - |this|.{{GPUCommandEncoder/[[state]]}} is {{encoder state/open}}.
+                - |querySet| is [$valid to use with$] |this|.
+                - |querySet|.{{GPUQuerySet/[[type]]}} is {{GPUQueryType/timestamp}}.
+                - |queryIndex| is less than |querySet|.{{GPUQuerySet/[[count]]}}.
+                - |queryIndex| cannot be overwritten on |this|.
```

Comment:
I think it would be better to explicitly have a sequence of booleans that represent whether a query is written or not and have the various query methods update it (or check it).

Reply:
It's only needed in native for the validation of end\*Query/writeTimestamp, query reset (Vulkan requires) and unused query resolve (Vulkan requires), users don't need to query or change the used query index, it should be updated by end\*Query/writeTimestamp/resolveQuerySet. 
Maybe we can add query state like *unavailable* and *available* to describe how the state changes and which state is required for each query method in the SPEC.

Reply:
Yes, the sequence of booleans would just be spec "private" like you described (not an added API).

Reply:
Oh, I missunderstood Corentin's comments. @kainino0x , thanks for your explaination.